airbyte-featureflag/src/main/kotlin/Context.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.featureflag
+
+/**
+ * Context abstraction around LaunchDarkly v6 context idea
+ *
+ * I'm still playing around with this.  Basically the idea is to define our own custom context types
+ * (by implementing this sealed interface) to ensure that we are consistently using the same identifiers
+ * throughout the code.
+ *
+ * @property [kind] determines the kind of context the implementation is,
+ * must be consistent for each type and should not be set by the caller of a context
+ * @property [key] is the unique identifier for the specific context, e.g. a user-id or workspace-id
+ */
+sealed interface Context {
```

Comment:
I am wondering if this implementation of `Context` is too restrictive.
When I am thinking of feature flags, eventually, it would be nice to be able to define a general context (User, Workspace, Connection, Source, Destination), and then in the feature flag definition, be able to say if user matches X, or if source is Y etc. Looking at this, it feels like context is currently only one dimension at a time.

Or maybe, adjusting the Client interface to work with a list of context could be a future extension.

Reply:
It should be working this way already (with the addition of the new `Context` types you called out).

It's up to the evaluator of the feature-flag itself to determine _given this feature-flag and this context, is this feature-flag enabled or not_, which is exactly what LaunchDarkly does.  

In LaunchDarkly you would configure _this_ feature-flag for _that_ User-Context to return true.   

For the `flags.yml` based client we would need to build the context support ourselves, and I'm hoping we can avoid that for as long as possible.